src/foam/dao/MaterializedDAO.js

```diff
@@ -88,7 +88,11 @@ foam.CLASS({
       type: 'Object',
       args: 'Object state, FObject value',
       javaCode: `
-        getDelegate().remove(adapt(value));
+        if ( getPredicate().f(value) ) {
+          var obj = adapt(value);
+          if ( obj != null )
+            getDelegate().remove(adapt(value));
+        }
```

Comment:
MDAO should already handle null.
```suggestion
          getDelegate().remove(adapt(value));
```